### PR TITLE
Remove USReceipt - Receipt design update FormRecognizer 

### PR DESCRIPTION
**Pull request by samvaity**
Closes: #11681

Initially, Java only had `FormField<T>` and with the current direction, we only want to support `FormField`. So removing `FormField<T>` and adding `FieldValue` and making the design more consistent with .NET.

### CODE REVIEW

File: sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/models/FieldValue.java [hunk 1]
```diff
@@ -0,0 +1,213 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (iscai-msft) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/models/FieldValue.java:1]: Why are you adding a public FieldValue to deal with the new receipt changes?

Comment (samvaity) [reply sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/models/FieldValue.java:1]: I only had the strongly typed `FormField<T>` which could hold any `T` but now we need a `FieldValue` to provide a `type` for conversion for the user.

Similar to [this](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormField.cs#L71).

File: sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/models/FieldValue.java [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+     * Set the integer value of the field.
+     *
+     * @param formFieldInteger the integer value of the field.
+     *
+     * @return the FieldValue object itself.
+     */
+    public FieldValue setFormFieldInteger(final Integer formFieldInteger) {
+        this.formFieldInteger = formFieldInteger;
+        return this;
+    }
```
Comment (kinelski) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/models/FieldValue.java:83]: How large this `Integer` can be? According to the service, we'll eventually start supporting 64-bits integers here (not a concern for Preview 3, I think).

File: sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/models/FormField.java [hunk 3]
```diff
@@ -40,14 +39,13 @@
 
     /**
      * Constructs a FormField object.
-     *
-     * @param confidence The confidence of the recognized field.
+     *  @param confidence The confidence of the recognized field.
      * @param labelText The label text value for the field.
```
Comment (kinelski) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/models/FormField.java:43]: I think this extra space before `@param` was an accident...?

File: sdk/formrecognizer/azure-ai-formrecognizer/CHANGELOG.md [hunk 4]
```diff
@@ -1,6 +1,9 @@
 # Release History
 
 ## 1.0.0-beta.3 (Unreleased)
+- Removed `USReceipt` and `USReceiptItem`.
+- Removed the extension class `ReceiptExtensions` that transformed the recognized receipt to `USReceipt`. 
+- Removed `ReceiptType` and the `receiptType` property from `RecognizedReceipt`.
```
Comment (kinelski) [inline sdk/formrecognizer/azure-ai-formrecognizer/CHANGELOG.md:6]: Did you mean `receiptLocale` property?

File: sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/models/FieldValue.java (hunk 2 above)
Comment (samvaity) [reply sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/models/FieldValue.java:83]: I think that should change the generated code for the service to start sending in a `Long` value so will defer doing this with the service update.

**Pull request merged**